### PR TITLE
Feature/socks and transparent modes

### PR DESCRIPTION
**Pull request by lups2000**
#### Description

In this PR, I've added the **Transparent** and **SOCKS** modes to the Advanced category. I was also considering whether it would be beneficial to implement a dropdown menu for the Advanced modes. This way, the Advanced options would be collapsed by default, and users could expand them by clicking a "+" icon. This could help save space for those who prefer to stick with the Recommended modes. What do you think?

Ref: #7063

https://github.com/user-attachments/assets/b71c8403-18d0-492e-93ca-4ecd83922c85

#### Checklist

 - [x] I have updated tests where applicable.
 - [ ] I have added an entry to the CHANGELOG.

### CODE REVIEW

File: web/src/js/components/Modes/Transparent.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+            />
+        );
+    });
+
+    return (
+        <div>
+            <h4 className="mode-title">Transparent Proxy</h4>
+            <p className="mode-description">
+                A transparent proxy routes traffic at the network layer without
+                client configuration.
```
Comment (mhils) [inline web/src/js/components/Modes/Transparent.tsx:28]: Replace with something like this?
> You [configure your routing table](https://docs.mitmproxy.org/stable/howto-transparent/) to send traffic through mitmproxy.

File: web/src/js/components/Modes/Socks.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            />
+        );
+    });
+
+    return (
+        <div>
+            <h4 className="mode-title">SOCKS Proxy</h4>
+            <p className="mode-description">
+                Similar the regular proxy mode, but using SOCKS5 instead of HTTP
+                for connection establishment with the proxy.
```
Comment (mhils) [inline web/src/js/components/Modes/Socks.tsx:28]: Or just:

> You manually configure your client application or device to use a SOCKSv5 proxy.

To match what we have for the regular proxy. What do you think? :)

File: web/src/js/components/Modes.tsx [hunk 3]
```diff
@@ -18,6 +20,11 @@ export default function Modes() {
                 {!platform.startsWith("linux") ? <Local /> : undefined}
                 <Wireguard />
                 <Reverse />
+            </div>
+            <h3>Advanced</h3>
+            <div className="modes-container">
+                <Transparent />
+                <Socks />
```
[the remark this answers is not in the document]
Comment (lups2000) [reply web/src/js/components/Modes.tsx:27]: ahaha okay 😅

File: web/src/js/components/Modes/Socks.tsx (hunk 2 above)
Comment (lups2000) [reply web/src/js/components/Modes/Socks.tsx:28]: yes yes it's fine for me, I just copied some text from the docs but I wanted to know your opinion about that :)

**Review by mhils — COMMENTED** [review]
We're missing a bit of coverage here because our test backendState is hardcoded to Windows (which now has no transparent mode).

Let's update https://github.com/mitmproxy/mitmproxy/blob/0f0c5ee250369476a689eea83e82b042c223ead0/web/gen/state_js.py#L48
to macOS (darwin) instead?

File: web/src/js/components/Modes/Transparent.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+                backendState={backendState[getSpec(server)]}
+            />
+        );
+    });
+
+    return (
+        <div>
+            <h4 className="mode-title">Transparent Proxy</h4>
+            <p className="mode-description">
+                You configure your routing table to send traffic through
```
[the remark this answers is not in the document]
Comment (lups2000) [reply web/src/js/components/Modes/Transparent.tsx:28]: yes sure, forgot it

**Pull request merged**